scalanet/src/io/iohk/scalanet/peergroup/dynamictls/DynamicTLSPeerGroupInternals.scala

```diff
@@ -113,30 +119,26 @@ private[dynamictls] object DynamicTLSPeerGroupInternals {
 
     private def handleEvent(event: ChannelEvent[M]): Unit =
       // Don't want to lose message, so `offer`, not `tryOffer`.
-      executeAsync(messageQueue.offer(event).void)
+      executeAsync(messageQueue.offer(channelConfig, event))
 
     private def executeAsync(task: Task[Unit]): Unit =
       task.runAsyncAndForget(scheduler)
   }
 
-  class ClientChannelImpl[M](
+  class ClientChannelBuilder[M](
       peerInfo: PeerInfo,
       clientBootstrap: Bootstrap,
       sslClientCtx: SslContext,
       framingConfig: DynamicTLSPeerGroup.FramingConfig,
+      maxIncomingQueueSize: Int,
       socks5Config: Option[Socks5Config]
   )(implicit scheduler: Scheduler, codec: Codec[M])
-      extends Channel[PeerInfo, M]
-      with StrictLogging {
+      extends StrictLogging {
     val (decoder, encoder) = buildFramingCodecs(framingConfig)
-
-    val to: PeerInfo = peerInfo
-
-    private val activation = Promise[io.netty.channel.Channel]()
+    private val to = peerInfo
+    private val messageQueue = makeMessageQueue[M](maxIncomingQueueSize)
```

Comment:
I see that `ch.channelConfig()` first becomes available in `initChannel`, but you need the `messageQueue` in `initialize`. So maybe, like the `activation`, this needs to be put in a `Deferred` or `Promise`, and created in `initChannel` where you can pass the `channelConfig` to its constructor. Then `initialize` can take it from there by awaiting the promise.